### ISSUE TITLE
Rule proposal: enforce if conditions to be boolean

### ISSUE THREAD

**Issue by fansanelli** [typescript-eslint/typescript-eslint#6898]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

To check that every if condition is boolean (or a function returning boolean).
Because testing an object, an array or a number may be confusing and lead unwanted behaviours.
This improves readability mostly, but may also avoid potential bugs, as done in other languages like Java.

### Fail Cases

```typescript
let a = 1;
if (a) do something...
```

### Pass Cases

```typescript
let a = 1;
if (a !== 0) do something...
```

### Additional Info

The same rule could be extended to while conditions too.

**Comment by bradzacher** [comment]
Check out our `strict-boolean-expressions` rule which cna be configured to do exactly this.

https://typescript-eslint.io/rules/strict-boolean-expressions/
